python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -182,7 +184,63 @@ def download_thumbnail(url, bundle):
 
         return path_to_cached_thumb
 
+    @staticmethod
+    def download_thumbnail_source(entity_type, entity_id, bundle):
+        """
+        Convenience and compatibility method for quick and easy synchronous thumbnail download.
+        This will retrieve the source file for a thumbnail given a shotgun entity type and id.
+        If the resolved thumbnail source file has already been cached, a path to it will be
+        returned instantly. Otherwise, it will be downloaded from Shotgun and placed in the
+        standard cache location on disk. The full path to cached thumbnail is returned. 
+
+        This is a helper method meant to make it easy to port over synchronous legacy
```

Comment:
nitpick: I suggest we add something here and to the old `download_thumbnail` method to point out that with the `download_thumbnail` you always get a jpeg with a fixed resolution and with the `download_thumbnail_source` you get whatever image (res and format) that was uploaded to sg.